### ISSUE TITLE
Notices: Allow link actions without requiring URLs

### ISSUE THREAD

**Issue by mmtr** [WordPress/gutenberg#13009]
**Is your feature request related to a problem? Please describe.**

We can define an `actions` prop for the `Notice` components in order to display actions related to the notice. This prop is an array of object containing 3 properties: `label`, `url` and `onClick`.

I realized that the action is only styled as a link if the `url` property is defined, but I think it should be possible to style it as a link even defining only the `onClick` property.

Observe how the "Learn more" action from the example below is not different from the rest of the notice message.

```jsx
const actions = [
	{
		label: 'Learn More',
		onClick: () => doSomethingWithoutRedirectToUrl()
	},
];

<Notice status="warning" actions={ actions }>
	To update, check your email and confirm your address.
</Notice>
```

<img width="559" alt="screen shot 2018-12-19 at 11 11 28" src="https://user-images.githubusercontent.com/1233880/50214844-8d6e6e80-0381-11e9-9787-871f5f7cb76f.png">

**Describe the solution you'd like**

Allow an additional property `isLink`, so the action can be styled as a link.

```jsx
const actions = [
	{
		label: 'Learn More',
		isLink: true,
		onClick: () => doSomethingWithoutRedirectToUrl()
	},
];

<Notice status="warning" actions={ actions }>
	To update, check your email and confirm your address.
</Notice>
```

<img width="453" alt="screen shot 2018-12-19 at 11 12 04" src="https://user-images.githubusercontent.com/1233880/50214922-d4f4fa80-0381-11e9-9b5f-a3aa4875fd2e.png">

**Describe alternatives you've considered**

Since the actions are styled as links if the `url` property is defined, I tried to use a hack where `url` is `true`, so the [`href` attribute of the `Button` component](https://github.com/WordPress/gutenberg/blob/170f2a8b2b3375134b1c880cc3592f7e1895c610/packages/components/src/notice/index.js#L42) is not set because `true` is not a valid value, but the [`isLink` prop](https://github.com/WordPress/gutenberg/blob/170f2a8b2b3375134b1c880cc3592f7e1895c610/packages/components/src/notice/index.js#L43) is `true`.

```jsx
const actions = [
	{
		label: 'Learn More',
		url: true,
		onClick: () => doSomethingWithoutRedirectToUrl()
	},
];

<Notice status="warning" actions={ actions }>
	To update, check your email and confirm your address.
</Notice>
```

This works, but causes a React warning because `true` is not a valid value for the `href` attribute.

```
Warning: Received `true` for a non-boolean attribute `href`.
```

**Comment by mmtr** [comment]
Just realized there is a workaround for avoiding the warning if the `url` value is `#` and the `onClick` callback executes `e.preventDefault()`:

```jsx
const actions = [
	{
		label: 'Learn More',
		url: '#',
		onClick: e => {
			e.preventDefault();
			doSomethingWithoutRedirectToUrl();
		},
	},
];

<Notice status="warning" actions={ actions }>
	To update, check your email and confirm your address.
</Notice>
```

Anyway, I still think it'd great to have an `isLink` option, so this work-around wouldn't be needed.

**Comment by afercia** [comment]
> the url value is #

If the action is `doSomethingWithoutRedirectToUrl`, i.e. not navigating to another page and "doing something" instead, then the notice should use a `<button>` element. Let's use button for buttons and links for links. For accessibility reasons, `href="#"` links should never be used.

**Comment by mmtr** [comment]
I agree with you @afercia! 🙂 

That's why I considered it a work-around and I still would like to see a way of defining buttons styled as links, or alternatively as primary buttons, so they can be differentiated from the notice message.

**Comment by afercia** [comment]
Looking a bit better, I now understand what you meant. Sorry, I'm a bit slow 😆 

So, omitting the `url` prop and passing the `onClick` one makes the action thing a button. However, the button is completely unstyled. I'd say this is a bug:

<img width="650" alt="screenshot 2018-12-27 at 10 35 49" src="https://user-images.githubusercontent.com/1682452/50475509-5e7d7b80-09c5-11e9-8bc3-35f78f2283cb.png">

At the very least, those buttons should be automatically styled as default buttons:

<img width="650" alt="screenshot 2018-12-27 at 10 40 05" src="https://user-images.githubusercontent.com/1682452/50475535-70f7b500-09c5-11e9-8075-0ed7d4a46814.png">

Also, I understand the need to style the button as primary, when necessary. So there should be an additional prop to pass class names.

From an accessibility perspective, I'm not so happy that's currently possible to pass both an `url` prop (so it's a link) _and_ an `onClick` prop: in this case it should be a button.

An option for your use case could be passing the buttons as children: seems Gutenberg does it for the template validation notice:

https://github.com/WordPress/gutenberg/blob/7ea50218a1de9cd00aceb0a58b4c879719f26d9b/packages/editor/src/components/template-validation-notice/index.js#L22-L28

but I understand that's less than ideal and this component should better handle the link / button actions.
